src/main/java/org/folio/circulation/support/AsyncCoordinationUtil.java

```diff
@@ -47,6 +48,29 @@ public static <T, R> CompletableFuture<List<Result<R>>> allResultsOf(
       .thenApply(v -> futures.stream().map(CompletableFuture::join).collect(Collectors.toList()));
   }
 
+  /**
+   * Applies BiFunction {@code asyncAction} to all key-value pairs in {@code map}
+   * and combines results to list
+   */
+  public static <T, R, S> CompletableFuture<Result<List<S>>> allOf(
+    Map<T, R> map, BiFunction<T, R, CompletableFuture<Result<S>>> asyncAction) {
+
+    return allResultsOf(map, asyncAction)
+      .thenApply(Result::combineAll);
+  }
+
+  private static <T, R, S> CompletableFuture<List<Result<S>>> allResultsOf(
+    Map<T, R> map, BiFunction<T, R, CompletableFuture<Result<S>>> asyncAction) {
+
+    List<CompletableFuture<Result<S>>> futures =
+      map.keySet().stream()
+        .map(key -> asyncAction.apply(key, map.get(key)))
```

Comment:
Maybe it's better to iterate over entry set? This way you we won't have to lookup value by key

Reply:
done